server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java

```diff
@@ -118,81 +100,50 @@ public void configure(HierarchicalConfiguration config) throws ConfigurationExce
      */
     @PostConstruct
     public void init() {
-        LOGGER.info("{} init...", getClass().getName());
-
+        LOGGER.info("init...");
         queue = queueFactory.createQueue(MailQueueFactory.SPOOL);
+        spooler = Schedulers.fromExecutor(Executors.newFixedThreadPool(numThreads, NamedThreadFactory.withName("spooler")));
+        LOGGER.info("uses {} Thread(s)", numThreads);
+    }
 
-        LOGGER.info("{} uses {} Thread(s)", getClass().getName(), numThreads);
+    public void run() {
+        LOGGER.info("Queue={}", queue);
 
-        active.set(true);
-        workerService = JMXEnabledThreadPoolExecutor.newFixedThreadPool("org.apache.james:type=component,component=mailetcontainer,name=mailspooler,sub-type=threadpool", "spooler", numThreads);
-        dequeueService = JMXEnabledThreadPoolExecutor.newFixedThreadPool("org.apache.james:type=component,component=mailetcontainer,name=mailspooler,sub-type=threadpool", "dequeuer", numDequeueThreads);
+        disposable = queue.deQueue()
+            .publishOn(spooler)
+            .flatMap(this::handleOnQueueItem)
+            .onErrorContinue((throwable, item) -> LOGGER.error("Exception processing mail while spooling {}", item, throwable))
+            .subscribeOn(Schedulers.elastic())
+            .subscribe();
+    }
 
-        for (int i = 0; i < numDequeueThreads; i++) {
-            Thread reader = new Thread(this, "Dequeue Thread #" + i);
-            dequeueService.execute(reader);
+    private Mono<Void> handleOnQueueItem(MailQueueItem queueItem) {
+        TimeMetric timeMetric = metricFactory.timer(SPOOL_PROCESSING);
+        try {
+            processingActive.incrementAndGet();
+            return processMail(queueItem);
+        } catch (Throwable e) {
```

Comment:
I do not see what you expect to catch, if you wait for `MailQueue.MailQueueException`, be a bit more precise, or deal with it in `processMail`.

Reply:
`MailQueueException` can be sent by the rollback (`done(false)`) so it's probably better to catch it here.
We also want to make sure that no exception will ever break the Flux as it has to live forever.
I'm not sure how to achieve that without a broad catch wrapping exceptions into an Mono.error.